napari/_qt/widgets/qt_viewer_buttons.py

```diff
@@ -26,6 +28,64 @@
     from ...viewer import Viewer
 
 
+def _spawn_new_image_widget(viewer: 'Viewer') -> None:
+    """
+    Creates a dialog to add an image to viewer.
+
+
+    Parameters
+    ----------
+    viewer : napari.components.ViewerModel
+        Napari viewer containing the rendered scene, layers, and controls.
+    """
+
+    # Define an enum for array type selection
+    class BackingData(Enum):
+        NumPy = 'NumPy'
+        Zarr = 'Zarr'
+
+    # Define the magicgui widget for parameter harvesting
+    @magicgui(
+        call_button="Create",
+        name={'tooltip': "If blank, a name will be generated"},
+        dimensions={'tooltip': "The size of the new image"},
+        array_type={'tooltip': "The backing data array implementation"},
+        fill_value={'tooltip': "Starting value for all pixels"},
+    )
+    def _new_image_widget(
+        name: str = "",
+        dimensions: List[int] = [512, 512],
```

Comment:
These values should be calculated base on the existing layer. You may look for inspiration here

https://github.com/napari/napari/blob/2dc5aa659f875c353bfbde3b20d8f07a664ed8a8/napari/components/viewer_model.py#L304-L316

Also, there should be scale parameters that should be possible to set/adjust.